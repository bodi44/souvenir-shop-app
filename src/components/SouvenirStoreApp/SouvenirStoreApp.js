import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import RequestForm from '../RequestForm'
import Output from '../Output'
import {
  getAllPurchasesByDate,
  getCurrencies,
  isCurrenciesLoading,
  isCurrenciesHasErrorLoading,
  isStoreHasErrorMessage,
  getTotalIncome, showHelpMessage
} from '../../reducers'
import { fetchCurrencies } from '../../actions/currencies'

import './SouvenirStoreApp.scss'
import BEM from '../../helpers/BEM'

const b = BEM('SouvenirStoreApp')

const SouvenirStoreApp = ({ fetchCurrencies, currencies, isLoading, hasError, allPurchases, storeErrorMessage, totalIncome, helpMessage }) => {
  useEffect(() => {
    fetchCurrencies()
  }, [fetchCurrencies])

  if (isLoading) return <div className={b('data-error')}>Data is Loading...</div>

  if (hasError) return <div className={b('data-error')}>Error occurred while fetching API</div>

  return (
    <div className={b()}>
      <h1 className={b('title')}>Souvenir-store App</h1>
      <RequestForm
        allPurchases={allPurchases}
        currencyData={currencies}/>
      <Output
        allPurchases={allPurchases}
        totalIncome={totalIncome}
        error={storeErrorMessage}
        helpMessage={helpMessage}/>
    </div>
  )
}

SouvenirStoreApp.propTypes = {
  currencies: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  hasError: PropTypes.any,
  allPurchases: PropTypes.array.isRequired,
  fetchCurrencies: PropTypes.func.isRequired,
  storeErrorMessage: PropTypes.string,
  totalIncome: PropTypes.string,
  helpMessage: PropTypes.bool.isRequired
}

export default connect(
  state => ({
    currencies: getCurrencies(state),
    isLoading: isCurrenciesLoading(state),
    hasError: isCurrenciesHasErrorLoading(state),
    allPurchases: getAllPurchasesByDate(state),
    storeErrorMessage: isStoreHasErrorMessage(state),
    totalIncome: getTotalIncome(state),
    helpMessage: showHelpMessage(state)
  }),
  {
    fetchCurrencies
  }
)(SouvenirStoreApp)