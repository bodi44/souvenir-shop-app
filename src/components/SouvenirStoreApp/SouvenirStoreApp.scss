.SouvenirStoreApp {
  margin: 0 auto;
  font-family: 'Roboto', sans-serif;
  display: flex;
  width: 50%;
  flex-direction: column;
  min-width: 325px;

  &__data-error {
    margin-top: 50px;
    font-size: 20px;
    display: block;
    vertical-align: middle;
    text-align: center;
  }

  &__title {
    display: block;
    text-align: center;
    margin: 50px auto;
  }
}