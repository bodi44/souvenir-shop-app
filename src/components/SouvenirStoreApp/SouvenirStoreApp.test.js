import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import SouvenirStoreApp from './SouvenirStoreApp'
import ProviderWrapper from '../Provider'
import configureStore from '../../configureStore'

Enzyme.configure({ adapter: new Adapter() })

describe('SouvenirStoreApp component', () => {
  it('should render correctly wrapped into Provider with no given props', () => {
    const component = shallow(
      <ProviderWrapper store={configureStore()}>
        <SouvenirStoreApp/>
      </ProviderWrapper>
    )
    expect(component).toMatchSnapshot()
  })
})