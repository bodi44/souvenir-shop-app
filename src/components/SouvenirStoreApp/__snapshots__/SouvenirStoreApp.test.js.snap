// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SouvenirStoreApp component should render correctly wrapped into Provider with no given props 1`] = `ShallowWrapper {}`;
